azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NetworkImpl.java

```diff
@@ -160,18 +160,19 @@ public NetworkImpl withAddressSpace(String cidr) {
     @Override
     protected void createResource() throws Exception {
         // Ensure address spaces
-        if(this.addressSpaces().size() == 0) {
+        if (this.addressSpaces().size() == 0) {
             this.withAddressSpace("10.0.0.0/16");
         }
 
         // Create a subnet as needed, covering the entire first address space
-        // TODO: this shouldn't happen during Update -- may need to move to initializer or something...
-        if(this.inner().subnets().size() == 0) {
+        // Future: this shouldn't happen during Update -- may need to move to initializer or something...
+        if (this.inner().subnets().size() == 0) {
             this.withSubnet("subnet1", this.addressSpaces().get(0));
         }
 
```

Comment:
you can use isInCreateMode() method here to ensure this happens only during create

```java
if (isInCreateMode() && this.inner().subnets().size() == 0) {
   this.withSubnet("subnet1", this.addressSpaces().get(0));
}
```